squidDrone/src/propulsion/propulsion_hardware_config.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef SRC_PROPULSION_PROPULSION_HARDWARE_CONFIG_HPP_
+#define SRC_PROPULSION_PROPULSION_HARDWARE_CONFIG_HPP_
+
+#include <cstdint>
+
+#include "propulsion_hardware_types.hpp"
+#include "stm32g4xx.h"
+
+namespace propulsion {
+
+/**
+ * @brief A type to accumulate all the info needed to build a Motor
+ * 
+ */
+struct PropulsionHardwareConfig{
+  /// Deleted default Ctor to forbid using this one
+  PropulsionHardwareConfig() = delete;
+
+  /// Custom Ctor to correclty construct this type
+  PropulsionHardwareConfig(types::MotorType motor_type, types::EscType esc_type, TIM_HandleTypeDef* timer, std::uint32_t channel)
+      : motor_type(motor_type), esc_type(esc_type), timer(timer), channel(channel) {}
+
+  /// MotorType enum defines the kind of motor used
+  types::MotorType motor_type;
+
+  /// EscType enum defines the kind of esc used
+  types::EscType esc_type;
+
+  /// A pointer to a HAL timer object. Simple pointer is used for future compatability with HAL
+  TIM_HandleTypeDef* timer;
+
+  /**
+   * @brief Channel as defined in HAL (see below)
+   *  TIM_CHANNEL_1 0x00000000U                          
```

Comment:
I also had this problem, to document values from HAL. If we generate HAL new or some of these values would change, than the documentation would be outdated. Maybe we should add HAL to doxygen?

Reply:
I think in this case it is ok, because the channel defines are the same for all ARM M4/7 cores.
But in general you are correct we should open an issue and add the current HAL to doxygen